src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php

```diff
@@ -32,7 +33,21 @@ public function isFresh($timestamp)
             return false;
         }
 
-        return @filemtime($this->filePath) <= $timestamp;
+        // has the file *not* been modified? Definitely fresh
+        if (@filemtime($this->filePath) <= $timestamp) {
+            return true;
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($this->class);
+        } catch (\ReflectionException $e) {
+            // the class does not exist anymore!
+            return false;
+        }
+
+        $newSelf = AutowirePass::createResourceForClass($reflectionClass);
+
+        return $this == $newSelf;
```

Comment:
In fact triple equals wouldn't fit as objects have different identities. Yet this is strange: @weaverryan shouldn't we `touch($this->filePath)` here when the objects are equals?
Would it be better to compare properties using strict comparisons? This could be done with:
`((array) $this === (array) AutowirePass::createResourceForClass($reflectionClass)`